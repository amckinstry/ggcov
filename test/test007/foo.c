#include <stdio.h>			    /* C(-) */
#include <stdlib.h>			    /* C(-) */
#include <setjmp.h>			    /* C(-) */
					    /* C(-) */
static jmp_buf jb;			    /* C(-) */
					    /* C(-) */
void
function_one(int x)
{
    printf("    function_one begins\n");    /* C(4) */
    if (!x)				    /* C(4) */
    	longjmp(jb, 1);			    /* C(1) */
    printf("    ...ends\n");		    /* C(3) */
}
					    /* C(-) */
void
function_two(int x)
{
    printf("    function_two begins\n");    /* C(3) */
    if (!x)				    /* C(3) */
    	longjmp(jb, 2);			    /* C(1) */
    printf("    ...ends\n");		    /* C(2) */
}
					    /* C(-) */
void
function_three(int x)
{
    printf("    function_three begins\n");  /* C(2) */
    if (!x)				    /* C(2) */
    	longjmp(jb, 3);			    /* C(1) */
    printf("    ...ends\n");		    /* C(1) */
}
					    /* C(-) */
int
main(int argc, char **argv)
{
    int x;
    
    printf("foo running\n");		    /* C(4) */
    x = atoi(argv[1]);			    /* C(4) */
    if (!setjmp(jb))
    {
	function_one(--x);		    /* C(4) */
	function_two(--x);		    /* C(3) */
	function_three(--x);		    /* C(2) */
	printf("function completed\n");     /* C(1) */
    }
    else
    {
	printf("long jump triggered\n");    /* C(3) */
    }
    x += 4;				    /* C(4) */
    return 0;				    /* C(4) */
}
