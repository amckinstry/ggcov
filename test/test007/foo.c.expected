        -:    1:#include <stdio.h>
        -:    2:#include <stdlib.h>
        -:    3:#include <setjmp.h>
        -:    4:
        -:    5:static jmp_buf jb;
        -:    6:
        -:    7:void
        -:    8:function_one(int x)
        4:    9:{
        4:   10:    printf("    function_one begins\n");
        4:   11:    if (!x)
        1:   12:    	longjmp(jb, 1);
        3:   13:    printf("    ...ends\n");
        -:   14:}
        -:   15:
        -:   16:void
        -:   17:function_two(int x)
        3:   18:{
        3:   19:    printf("    function_two begins\n");
        3:   20:    if (!x)
        1:   21:    	longjmp(jb, 2);
        2:   22:    printf("    ...ends\n");
        -:   23:}
        -:   24:
        -:   25:void
        -:   26:function_three(int x)
        2:   27:{
        2:   28:    printf("    function_three begins\n");
        2:   29:    if (!x)
        1:   30:    	longjmp(jb, 3);
        1:   31:    printf("    ...ends\n");
        -:   32:}
        -:   33:
        -:   34:int
        -:   35:main(int argc, char **argv)
        4:   36:{
        -:   37:    int x;
        -:   38:    
        4:   39:    printf("foo running\n");
        4:   40:    x = atoi(argv[1]);
        7:   41:    if (!setjmp(jb))
        -:   42:    {
        4:   43:	function_one(--x);
        3:   44:	function_two(--x);
        2:   45:	function_three(--x);
        -:   46:    }
        4:   47:    x += 4;
        4:   48:    printf("foo returning naturally\n");
        4:   49:    return 0;
        -:   50:}
